### DISCUSSION TITLE
Set variable with script part

### DISCUSSION THREAD

**Discussion by LindevdH** [AnWeber/vscode-httpyac#172] [Q&A] [answered]
On the base of programming I see my self as a learner/newbe.
With VSCode I am performing checks and manipulations on api calls. At this moment I have the following setup:

```
@ip = --.-.--.--
@port = ---1
@company = ***
@stable = 2
@flocknumber = 0


### GETBasics
# @name Basics
GET http://{{ip}}:{{port}}/basics/{{company}}/{{company}}-{{stable}}


### Get Medication
# @name medication
# @ref Basics

{{
    if( $global.flocknumber === 0) {
        $global.flocknumber = Basics[0].flockNumber;
    } 
}}
    
GET http://{{ip}}:{{port}}/medication/{{flocknumber}} 

```
I am Trying to change the variable @flocknumber into the the value of the referenced api call @Basics if it is equal to 0.
At this point I don't get any errors The only thing that I can Trace is that ther Get Medication url is executed with a 0 for flocknumber.

**Comment by AnWeber** [comment]
I think that the variable flock number is handled as string and you use `===`. Try using `==` or convert the variable to a number

**Reply by LindevdH** [reply]
Unfortunalty that did not make any change.

**Reply by AnWeber** [reply]
All variables can be accessed using just the name,  `flocknumber`.  But all variables are scoped to current context. `global` can be used to prevent the scoping behavior. 
I will create a working example,but my notebook is really farm away at he.moment:-)

**Reply by AnWeber** [reply]
I did forget to mention the needed exports of the flocknumber. 
```
@flocknumber = 0

###

### GETBasics
# @name Basics
GET https://httpbin.org/json


### Get Medication
# @name medication
# @ref Basics

{{
    console.info(flocknumber, typeof flocknumber);
    if( flocknumber === "0") {
        exports.flocknumber = Basics.slideshow.slides[0].title;
    } 
}}



GET https://httpbin.org/anything?flock={{flocknumber}}
```

**Comment by LindevdH** [comment, marked as answer]
@AnWeber  Thanks for the reply/example.
I was able to test the solution today. My final script:

```
@ip = --.-.--.--
@port = ---1
@company = ***
@stable = 2
@flocknumber = 0

### GETBasics
# @name Basics
GET http://{{ip}}:{{port}}/basics/{{company}}/{{company}}-{{stable}}

### Get Medication
# @name medication
# @ref Basics

{{
    console.info(flocknumber, typeof flocknumber);
    if( flocknumber === "0") {
        exports.flocknumber = Basics[0].flockNumber;
    } 
}}
    
GET http://{{ip}}:{{port}}/medication/{{flocknumber}} 
```
This wil help me to setup manipulation scripts. 
I will have to look for a way to clear the cache in order to work with changes without the need of opening and closing VScoding app. But that is another question.  
Thanks for the help.  Your solution is realy nice and helpfull.

Kind regard.
Harm van der Linde

**Reply by LindevdH** [reply]
I was not able to mark the reply of @AnWeber  as answer. Otherwise I would have done that.

**Comment by AnWeber** [comment]
You don`t need to close VSCode. Use command `httpyac.reset`. It clears all my caches
